src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -342,11 +327,135 @@ public async Task<ActionResult> Post(string appId, int? instanceOwnerId, [FromBo
             }
             catch (Exception e)
             {
-                logger.LogError($"Unable to create {appId} instance for {theInstanceOwnerId} due to {e}");
-                return StatusCode(500, $"Unable to create {appId} instance for {theInstanceOwnerId} due to {e}");
+                logger.LogError($"Unable to create {appId} instance for {ownerId} due to {e}");
+                return StatusCode(500, $"Unable to create {appId} instance for {ownerId} due to {e}");
             }
         }
 
+        private Application GetApplicationOrError(string appId, out ActionResult errorResult)
+        {
+            errorResult = null;
+            Application appInfo = null;
+
+            try
+            {
+                string org = appId.Split("/")[0];
+
+                appInfo = _applicationRepository.FindOne(appId, org).Result;                
+            }
+            catch (DocumentClientException dce)
+            {
+                if (dce.Error.Code.Equals("NotFound"))
+                {
+                    errorResult = NotFound($"Did not find application with appId={appId}");
+                }
+                else
+                {
+                    errorResult = StatusCode(500, $"Document database error: {dce}");
+                }
+            }
+            catch (Exception e)
+            {
+                errorResult = StatusCode(500, $"Unable to perform request: {e}");
+            }
+
+            return appInfo;
+        }
+
+        /// <summary>
+        /// InstanceOwner can be given in three different ways:
+        ///  - instanceOwnerId is provided as query param (priority1),
+        ///  - in instanceTemplate.instanceOwnerId (priority2),
+        ///  - or instanceTemplate.instanceOwnerLookup (priority3)
+        /// </summary>
+        /// <param name="instanceOwnerId">the instance owner id</param>
+        /// <param name="instanceTemplate">the instance template</param>
+        /// <param name="errorResult">the errorResult. null if successful otherwise an action result</param>
+        /// <returns></returns>
+        private int GetOrLookupInstanceOwnerId(int? instanceOwnerId, Instance instanceTemplate, out ActionResult errorResult)
+        {
+            errorResult = null;
+            int ownerId = instanceOwnerId ?? 0;           
```

Comment:
this operation can be avoided by checking instanceOwnerId != null in the if statement below. As the value is simply rewritten at a later point is there a need to assign it this early?

Reply:
OK